docs/spec_draft.md

```diff
@@ -664,6 +665,143 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dot_general
+
+### Semantics
+
+Performs general dot product between `lhs` and `rhs` over dimensions specified
+in lhs/rhs batching and contracting dimensions and produces a `result` tensor.
+
+This general form allows dot product between multi-dimensional tensors, and to
+contract multiple dimensions at a time. It also allows to specify the batching
+dimensions, which determines the dimensions to avoid summing during contraciton.
+
+More formally, let:
+* `batching_dims` = [0, 1, ..., size(`lhs_batching_dims`)].
+* LCD = `lhs_contracting_dimensions`.
+* RCD = `rhs_contracting_dimensions`.
+
+$$result[...,\ ip,\ ...] = \sum_{d\ =\ 0}^{size(LCD)} \sum_{(c,\ c')\ =\
+(0,\ 0)}^{(dim(lhs,\ LCD[d]),\ dim(rhs,\ RCD[d]))}
+lhs(...,\ jc,\ ...,\ jq,\ ...)\ *\ rhs(...,\ kc',\ ...,\ kr,\ ...)$$
+
+* If p is in `batching_dims`, j[`lhs_batching_dims`[p]] = ip and
+k[`rhs_batching_dims`[p]] = ip.
+* jq = kr = ip otherwise.
+
+`precision_config` controls the tradeoff between speed and accuracy for
+computations on accelerator backends. This can be one of the following:
+
```

Comment:
You can add the constraints on precision config following the  changes in 
https://github.com/openxla/stablehlo/pull/441